src/utils.cpp

```diff
@@ -735,55 +736,53 @@ std::vector<std::pair<std::string, std::string>> get_cgroup_paths(
     uint64_t cgroupid,
     std::string filter)
 {
-  // TODO: Rewrite using std::views when C++20 support becomes common
   auto roots = get_cgroup_hierarchy_roots();
 
   // Replace cgroup version with cgroup mount point directory name for cgroupv1
   // roots and "unified" for cgroupv2 roots
-  for (auto &root : roots) {
-    if (root.first == "cgroup") {
-      root = { std::filesystem::path(root.second).filename().string(),
-               root.second };
-    } else if (root.first == "cgroup2") {
-      root = { "unified", root.second };
-    }
-  }
+  auto types = roots |
+               std::views::transform(
+                   [&](auto &root) -> std::pair<std::string, std::string> {
+                     if (root.first == "cgroup") {
+                       return {
+                         std::filesystem::path(root.second).filename().string(),
+                         root.second
+                       };
+                     } else if (root.first == "cgroup2") {
+                       return { "unified", root.second };
+                     }
+                     LOG(BUG) << "Unknown cgroup filesystem: " << root.first;
+                     __builtin_unreachable();
```

Comment:
Maybe we should be throwing here so we just have a single statement?

Reply:
I actually just changed the structure here to eliminate the need for any kind of branch & assertion in this path.

Reply:
> This is a bigger thread, I will file a separate bug.

Definitely 😄  Saw the new thread. Thanks for filing and thinking about! I'll comment there.